python/lsst/ts/FiberSpectrograph/avsFiberSpectrograph.py

```diff
@@ -316,9 +316,13 @@ def decode(value):
             readout from the device.
         asyncio.CancelledError
             Raised if the exposure is stopped before it is read out.
+        ValueError
+            Raised if the exposure time is outside of the valid range.
         """
         if not self._expose_task.done():
             raise RuntimeError("Cannot start new exposure until current exposure finishes.")
+        if (duration < 0.000002) or (duration > 600):
+            raise ValueError("Exposure duration not in valid range: 0.002ms - 600s")
```

Comment:
Consider using constants so the test and message use the same data. Also if you make them publicly available you can test for values around the limits.

Reply:
Consider writing 0.000002 as 0.000_002 for clarity.

Reply:
I suggest using seconds in the message (perhaps with scientific notation) since that's the units required by the function (and appropriately so, since all data comes from DDS messages that use seconds).

Reply:
Done: python's default formatting writes "2e-06" for the minimum time, which is perfect.